Tests/NIOHTTP2Tests/TestUtilities.swift

```diff
@@ -63,6 +63,60 @@ extension XCTestCase {
         } while operated
     }
 
+    /// Have two `EmbeddedChannel` objects send and receive data from each and throw errors other until
+    /// they make no forward progress.
+    func interactInMemoryExpectingErrors(
```

Comment:
There's a lot of duplication here. Can we remove some of it?

We could (probably) have a single function:

```swift
func interactInMemory(
  _ first: EmbeddedChannel,
  _ second: EmbeddedChannel,
  expectError: Bool = false,
  file: StaticString = #filePath,
  line: Uint = #line,
  _ errorHandler: (_ error: any Error) -> Void = { _ in }
) {
  // ...
}
```